app/models/user.rb

```diff
@@ -1,16 +1,32 @@
+class Error < StandardError
+end
+
 class User < ApplicationRecord
   validates :name, presence: true, uniqueness: true
   validates :email, uniqueness: true, allow_blank: true, format: {
     with: EMAIL_REGEX,
     message: 'has invalid format',
   }
+
   has_secure_password
   after_destroy :ensure_a_user_remains
+  after_create :send_welcome_email
```

Comment:
why `after_create` and why not `after_create_commit` ?
Let me know the difference between the two and which one to use when